### PR TITLE
New FS test trial

### PR DESCRIPTION
**Pull request by olliecurtis**
This PR is the new basis for the new full stack recruitment test that will be piloted, before proceeding ensure you have read the Proposal document 😄 

Contact me if you have any issues or concerns

### CODE REVIEW

File: FOLLOW-UP.md [hunk 1]
```diff
@@ -4,18 +4,16 @@
 
 ### Q) What is the command to start the server?
```
Comment (shaundon) [inline FOLLOW-UP.md]:
```suggestion
### Q) What's the command to start the local dev environment?
```

I've been meaning to change this for ages, could we make it something that covers the whole test? I've included a suggestion but we don't need to make it exactly that, just something along the same lines.

File: README.md [hunk 2]
```diff
@@ -46,57 +40,40 @@ We've set you up with a build based on [our custom fork](https://backpack.github
...
+`npm start`
 
 ## Submission Guidelines
 
-* Please submit your program by sending a zip file to your Skyscanner contact.
-
+### `TODO: Update with the hired platform submission steps`
 * The zip file should be named {yourname}.zip, and should itself contain the full-stack-recruitment-test project folder with your submission.
 
 * The zip file should contain the [FOLLOW-UP.md](./FOLLOW-UP.md) file with answers to the follow-up questions.
```
Comment (mrkalndz) [inline README.md:74]: Should we give guidelines on how much time/content we expect in follow_up.md? If we're moving candidates towards a soft time limit and we are still grading on follow_up.md to me it feels sensible at least for us to have an opinion on how long we expect candidates to spend filling this in. Something like

_We would like to know the reasoning behind the technical choices you made in completing the project, please spend around 10 minutes completing follow_up.md_

File: FOLLOW-UP.md [hunk 3]
```diff
@@ -4,18 +4,16 @@
```
Comment (matthewdavidson) [inline FOLLOW-UP.md:4]: I think we should get rid of the follow up entirely. It's an extra part that is covered later in the recruitment process (rational for technical choices, depth of expertise etc).

**Review by matthewdavidson — COMMENTED** [review]
The readme in the `client/` folder is quite big and, while useful, could confuse candidates and waste valuable time for them if they think they have to read it in it's entirety.

File: README.md (hunk 2 above)
Comment (olliecurtis) [reply README.md:74]: Yes we should add something like this to give a rough estimate, I need to also check how this works with the hired platform in this scenario to see if it goes here or direct on the platform 😄

File: FOLLOW-UP.md (hunk 3 above)
Comment (olliecurtis) [reply FOLLOW-UP.md:4]: Maybe we shrink it to not cover so many requirements to get some understanding on choice made e.g. why not Backpack over other libs used (could be due to docs or something else) or choices on adding libs that may not be necessary, afterall its something we grade on and would be good to get test feedback.

Can always trial having it and then remove it if the answers aren't giving much extra value and we can give a guide of approx 10 mins to write a quick follow up on the test overall

File: client/src/index.js [hunk 4]
```diff
@@ -10,16 +9,7 @@ import App from './components/App';
...
-
-fetch('http://localhost:4000/api/search')
-  .then(response => response.json())
-  .then((results) => {
-    console.log('TODO: something with these results:');
-    console.log(results);
-  })
-  .catch(console.error);
-
+/* Create fuctionality in a sensible file to od the following:
```
Comment (matthewdavidson) [inline client/src/index.js]:
```suggestion
/* Create functionality in a sensible file to do the following:
```

File: package.json [hunk 5]
```diff
@@ -8,12 +8,10 @@
   "version": "1.0.0",
   "description": "Skyscanner full-stack recruitment test.",
   "scripts": {
-    "install": "(cd ./client && npm install) && (cd ./server && npm install)",
-    "client": "(cd ./client && npm start)",
+    "install": "(cd ./client && npm install)",
```
Comment (matthewdavidson) [inline package.json]: If we have removed the server aspect - we should get rid of the `client/` folder. This top level `package.json` is redundant now

**Review by matthewdavidson — COMMENTED** [review]
The candidate will struggle to fetch the `results.json` dude to CORS issues btw. We need to come up with a different approach. Perhaps we include it in the `public/` folder ?

**Comment by olliecurtis** [comment]
> The candidate will struggle to fetch the `results.json` dude to CORS issues btw. We need to come up with a different approach. Perhaps we include it in the `public/` folder ?

@matthewdavidson - That sounds like a sensible idea, I wonder how mobile get around this? But this will be a good easier place to work with.

**Comment by matthewdavidson** [comment]
@ojcurtis this isn't a concern for mobile - CORS is a browser mechanism.

**Review by georgegillams — APPROVED** [review]

**Pull request merged**